From fc2b1f8e919a9f4c9064348798f4140f54f6c209 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Fri, 5 Jan 2018 19:37:14 +0800
Subject: [PATCH 032/116] arm64: allwinner: h6: add basical DVFS support

Add basical DVFS support for Allwinner H6 SoC.

Only one lowest OPP is added, as its highest voltage is under
the initial voltage of AXP805's DCDCA regulator (0.9V).

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 arch/arm64/boot/dts/allwinner/sun50i-h6.dtsi | 39 ++++++++++++++++++++
 1 file changed, 39 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h6.dtsi b/arch/arm64/boot/dts/allwinner/sun50i-h6.dtsi
index 41b58ffa85..d10aa267b3 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h6.dtsi
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h6.dtsi
@@ -16,6 +16,21 @@
 	#address-cells = <1>;
 	#size-cells = <1>;
 
+	cpu0_opp_table: opp_table0 {
+		compatible = "allwinner,sun50i-h6-operating-points";
+		nvmem-cells = <&speedbin>;
+		opp-shared;
+
+		opp-888000000 {
+			opp-hz = /bits/ 64 <888000000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+
+			opp-microvolt-speed0 = <880000>;
+			opp-microvolt-speed1 = <820000>;
+			opp-microvolt-speed2 = <800000>;
+		};
+	};
+
 	cpus {
 		#address-cells = <1>;
 		#size-cells = <0>;
@@ -25,6 +40,11 @@
 			device_type = "cpu";
 			reg = <0>;
 			enable-method = "psci";
+
+			clocks = <&ccu CLK_CPUX>;
+			clock-names = "cpu";
+			operating-points-v2 = <&cpu0_opp_table>;
+			#cooling-cells = <2>;
 		};
 
 		cpu1: cpu@1 {
@@ -32,6 +52,11 @@
 			device_type = "cpu";
 			reg = <1>;
 			enable-method = "psci";
+
+			clocks = <&ccu CLK_CPUX>;
+			clock-names = "cpu";
+			operating-points-v2 = <&cpu0_opp_table>;
+			#cooling-cells = <2>;
 		};
 
 		cpu2: cpu@2 {
@@ -39,6 +64,11 @@
 			device_type = "cpu";
 			reg = <2>;
 			enable-method = "psci";
+
+			clocks = <&ccu CLK_CPUX>;
+			clock-names = "cpu";
+			operating-points-v2 = <&cpu0_opp_table>;
+			#cooling-cells = <2>;
 		};
 
 		cpu3: cpu@3 {
@@ -46,6 +76,11 @@
 			device_type = "cpu";
 			reg = <3>;
 			enable-method = "psci";
+
+			clocks = <&ccu CLK_CPUX>;
+			clock-names = "cpu";
+			operating-points-v2 = <&cpu0_opp_table>;
+			#cooling-cells = <2>;
 		};
 	};
 
@@ -216,6 +251,10 @@
 					reg = <0x000000 0x200000>;
 				};
 			};
+
+			speedbin: speed-bin@1c {
+				reg = <0x1c 4>;
+			};
 		};
 
 		ccu: clock@3001000 {
-- 
2.21.0

